src/Middleware/auth/auth.guard.ts

```diff
@@ -2,56 +2,45 @@ import { Injectable, NestMiddleware } from '@nestjs/common';
 import { Request, Response, NextFunction } from 'express';
 import * as firebase from 'firebase-admin';
 import { FirebaseApp } from '../../Firebase/firebase.service';
-// import * as serviceAccount from '../../Firebase/firebaseConfig.json';
+import { auth } from 'firebase-admin';
+import UserRecord = auth.UserRecord;
 
-// const firebase_params: object = {
-//   type: serviceAccount.type,
-//   project_id: serviceAccount.project_id,
-//   private_key_id: serviceAccount.private_key_id,
-//   private_key: serviceAccount.private_key,
-//   client_email: serviceAccount.client_email,
-//   client_id: serviceAccount.client_id,
-//   auth_uri: serviceAccount.auth_uri,
-//   token_uri: serviceAccount.token_uri,
-//   auth_provider_x509_cert_url: serviceAccount.client_x509_cert_url,
-//   client_x509_cert_url: serviceAccount.client_x509_cert_url,
-//   universe_domain: serviceAccount.universe_domain,
-// };
 @Injectable()
 export class PreauthMiddleware implements NestMiddleware {
   private auth: firebase.auth.Auth;
-  constructor(private firebaseApp: FirebaseApp) {
+  constructor(private readonly firebaseApp: FirebaseApp) {
     this.auth = firebaseApp.getAuth();
   }
-  use(req: Request, res: Response, next: NextFunction) {
-    console.log('Request...');
+  async use(req: Request, res: Response, next: NextFunction) {
+    // console.log('Request...');
     const token = req.headers.authorization;
-    // console.log(token);
-    if (token != null && token != '') {
+    if (token !== null && token !== '') {
       // console.log(token);
-      this.auth
-        // .auth()
-        .verifyIdToken(token.replace('Bearer ', ''))
-        .then(async (decodedToken) => {
-          // console.log(decodedToken);
-          const user: object = {
-            name: decodedToken.name,
-            email: decodedToken.email,
-            picture: decodedToken.picture,
-            user_id: decodedToken.user_id,
-          };
-          req['user'] = user;
-          next();
-        })
-        .catch((error): void => {
-          console.error(error);
-          this.accessDenied(req.url, res);
-        });
+      try {
+        const decodedToken = await this.auth.verifyIdToken(
+          token.replace('Bearer ', ''),
+        );
+        console.log(decodedToken);
```

Comment:
Please have lint rule for avoiding pushing `console.log`. With separate PR we can introduce husky as well

Reply:
all consoles and comments are removed now